src/zarr/testing/strategies.py

```diff
@@ -30,17 +32,17 @@
 )
 
 
-@st.composite  # type: ignore[misc]
-def keys(draw: st.DrawFn, *, max_num_nodes: int | None = None) -> Any:
+@st.composite
+def keys(draw: st.DrawFn, *, max_num_nodes: int | None = None) -> str:
     return draw(st.lists(node_names, min_size=1, max_size=max_num_nodes).map("/".join))
 
 
-@st.composite  # type: ignore[misc]
-def paths(draw: st.DrawFn, *, max_num_nodes: int | None = None) -> Any:
+@st.composite
+def paths(draw: st.DrawFn, *, max_num_nodes: int | None = None) -> str:
     return draw(st.just("/") | keys(max_num_nodes=max_num_nodes))
```

Comment:
```suggestion
    return draw(st.none() | st.just("/") | keys(max_num_nodes=max_num_nodes))
```

If `path=None` is allowed.